src/libs/OptionsListUtils.ts

```diff
@@ -2458,20 +2502,35 @@ function filterOptions(options: Options, searchInputValue: string, config?: Filt
         };
     }, options);
 
-    const recentReports = matchResults.recentReports.concat(matchResults.personalDetails);
-    const userToInvite =
-        recentReports.length === 0
-            ? getUserToInviteOption({
-                  searchValue,
-                  betas,
-              })
-            : null;
+    let {recentReports, personalDetails} = matchResults;
+
+    if (sortByReportTypeInSearch) {
+        recentReports = recentReports.concat(personalDetails);
+        personalDetails = [];
+        recentReports = orderOptions(recentReports, searchValue);
+    }
+
+    let userToInvite = null;
+    if (canInviteUser) {
+        if (recentReports.length === 0) {
```

Comment:
@TMisiukiewicz @rinej This causes a regression https://github.com/Expensify/App/issues/44200 where due to the default `sortByReportTypeInSearch` is false which won't concat `recentReports` & `personalDetails`. With that even though we have entries in `personalDetails` it will add that entry to `userToInvite` as well because of this truthy condition.

I think we also need to check the `personalDetails.length` here.

Reply:
opened a PR #44253 with a fix

Reply:
Thanks! I will check it.